R/functions.R

```diff
@@ -25,15 +25,23 @@ get_data <- function(synID, version = NULL){
 #'@param factors A vector of factor variables
 #'@param continuous A vector of continuous variables
 #' Remove primary variable for now - A vector of primary variables
-#'@param sample_variable A character vector with the column name corresponding to sample identifiers
 #'
 #'@export
 #'@return A data frame with coerced variables.
 #'@examples
-clean_covariates <- function(md, factors, continuous, sample_variable){
-  md[,factors] <- lapply(md[,factors], factor)
-  md[,continuous] <- lapply(md[,continuous], as.numeric)
-  md
+clean_covariates <- function(md, factors, continuous){
+  if (missing(factors) | missing(continuous)) {
+    stop("Factor and continuous variables are required.")
+  } else if (length(intersect(factors, continuous)) != 0) {
+    stop("Variables are present in both the continuous and factor arguments. Variables must be designated
+         numeric or factor, not both.")
+  } else if (!(factors %in% colnames(md) | !(continuous %in% colnames(md)))) {
+    stop("Variables provided are not present in the metadata.")
+  } else {
+    md[,factors] <- lapply(md[,factors], factor)
```

Comment:
```suggestion
    md[,factors] <- lapply(md[,factors, drop = FALSE], factor)
```

Recommend adding `drop = FALSE` to ensure that the result of subsetting `md` is always a data.frame. Otherwise, if there is only one factor column, then `md[, factors]` returns a vector, and `lapply()` will iterate over the values in the vector, producing incorrect results:

```r
dat <- data.frame(x = c("a", "b"), y = c("c", "d"))
clean_covariates(dat, factors = "x", continuous = "y")
#>   x y
#> 1 a 1
#> 2 a 1
#> Warning messages:
#> 1: In `[<-.data.frame`(`*tmp*`, , factors, value = list(1L, 1L)) :
#>   provided 2 variables to replace 1 variables
#> 2: In `[<-.data.frame`(`*tmp*`, , continuous, value = list(1, 2)) :
#>   provided 2 variables to replace 1 variables
```

Reply:
GOOD catch!